src/main/java/com/icthh/xm/uaa/service/otp/OtpService.java

```diff
@@ -0,0 +1,137 @@
+package com.icthh.xm.uaa.service.otp;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.icthh.xm.commons.exceptions.BusinessException;
+import com.icthh.xm.uaa.domain.UserLoginType;
+import com.icthh.xm.uaa.security.DomainUserDetails;
+import com.icthh.xm.uaa.service.TenantPropertiesService;
+import com.icthh.xm.uaa.service.dto.UserLoginDto;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.NotImplementedException;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.icthh.xm.uaa.config.Constants.TOKEN_AUTH_DETAILS_TFA_DESTINATION;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class OtpService {
```

Comment:
Please check the logs generated by this class and probably mask OTP value from log using `@LoggingAspectConfig` annnotation.